samples/subsys/usb/hid-mouse/src/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018 qianfan Zhao
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <board.h>
+#include <device.h>
+#include <gpio.h>
+
+#include <usb/usb_device.h>
+#include <usb/class/usb_hid.h>
+
+/* change this to use another GPIO port */
+#ifdef SW0_GPIO_NAME
+#define PORT	SW0_GPIO_NAME
+#else
+#error SW0_GPIO_NAME needs to be set in board.h
+#endif
+
+/* change this to use another GPIO pin */
+#ifdef SW0_GPIO_PIN
+#define PIN     SW0_GPIO_PIN
+#else
+#error SW0_GPIO_PIN needs to be set in board.h
+#endif
+
+/*
+ * Usage Page (Desktop),              ; Generic desktop controls (01h)
+ * Usage (Mouse),                     ; Mouse (02h, application collection)
+ * Collection (Application),
+ *    Usage (Pointer),                ; Pointer (01h, physical collection)
+ *    Collection (Physical),
+ *        Usage Page (Button),        ; Button (09h)
+ *        Usage Minimum (01h),
+ *        Usage Maximum (03h),
+ *        Logical Minimum (0),
+ *        Logical Maximum (1),
+ *        Report Count (3),
+ *        Report Size (1),
+ *        Input (Variable),
+ *        Report Count (1),
+ *        Report Size (5),
+ *        Input (Constant),
+ *        Usage Page (Desktop),       ; Generic desktop controls (01h)
+ *        Usage (X),                  ; X (30h, dynamic value)
+ *        Usage (Y),                  ; Y (31h, dynamic value)
+ *        Usage (Wheel),              ; Wheel (38h, dynamic value)
+ *        Logical Minimum (-127),
+ *        Logical Maximum (127),
+ *        Report Size (8),
+ *        Report Count (3),
+ *        Input (Variable, Relative),
+ *    End Collection,
+ * End Collection
+ */
+
+static const u8_t hid_report_desc[] = {
```

Comment:
Why don't you use the macros defined in `usb/class/usb_hid.h`?
`hid_report_desc[]` in samples/subsys/usb/hid looks cleaner and easier to read.

Reply:
but i think the comment is easier to read, at least for me it is. actually i can't know whats the  rules write hid report , that's report was copy from my thinkpad mouse, and translated from hex code to comment by using 'hidrd' tools .